### PR TITLE
[mono] Remove CoreFX.issues_windows.rsp file

### PR DESCRIPTION
**Pull request by MaximLipnin**
Changes:
- enabled `runtime (Libraries Test Run release mono Windows_NT x64 Debug)` CI lane (Windows Nano Server CI leg is disabled);
- enabled the tests that pass;
- marked failing test assemblies/classes/methods with `ActiveIssue` attribute using respective GH issues;
- removed CoreFX.issues_windows.rsp file and related mono rsp arguments.

  

Part of #1980

### CODE REVIEW

**Comment by MaximLipnin** [comment]
https://github.com/dotnet/runtime/issues/33549

**Comment by MaximLipnin** [comment]
@ViktorHofer Could you please help here? I'm trying to enable `runtime (Libraries Test Run release mono Windows_NT x64 Debug)` lane at least with the current rsp file, yet it still fails. Not sure if it's the cause but all the logs contain the following:
```
Failed to load the dll from [C:\helix\work\correlation\shared\Microsoft.NETCore.App\5.0.0\coreclr.dll], HRESULT: 0x8007007E
Failed to bind to CoreCLR at 'C:\helix\work\correlation\shared\Microsoft.NETCore.App\5.0.0\'
Failed to create CoreCLR, HRESULT: 0x80008088
----- end Mon 03/16/2020 20:59:12.03 ----- exit code -2147450743 ----------------------------------------------------------
```

**Comment by MaximLipnin** [comment]
According to @lateralusX, the possible cause is that:

> Windows mono runtime has a dependency on c-runtime DLL, and if a different machine is running tests compare to build machine that redist needs to be installed and available, or you will get failures to load coreclr.dll due to missing dependencies. CoreCLR uses static c-runtime dependency ..., but Mono in the past always used dynamic linked c-runtime (since that has been a recommendation for ages).

**Comment by lateralusX** [comment]
I will do the change to build using static c-runtime for Mono runtime since that was in the plan aligning to CoreCLR redist requirements. Since the failing lane is running debug Mono runtime, I strongly believe issue is related to this, since debug c-runtime is seldom available on host that don't have full toolchain installed.

**Comment by lateralusX** [comment]
https://github.com/dotnet/runtime/pull/33715

**Comment by ViktorHofer** [comment]
@MaximLipnin sounds your are unblocked?

**Comment by MaximLipnin** [comment]
@ViktorHofer Unfortunately, no. According to the latest log, there are over 400 errors, and the crashes mentioned above still occur. Perhaps, @lateralusX will try another fix for that (in addition to what he has already done).

**Comment by lateralusX** [comment]
CoreCLR seems to static link vcruntime140.dll even in release builds (not default behaviour when using /MD compiler switch). Looking at the install package that library is not included and since we still need it, I believe that is why we still fail to load, due to missing dependency. Problem is not seen if the c-runtime redist exists and I believe CI don't have that installed. I will try to do same thing as CoreCLR does static link vcruntime140.dll to reduce dependencies not included in package.

**Comment by lateralusX** [comment]
This should get rid of the vcruntime140.dll runtime dependecy:

https://github.com/dotnet/runtime/pull/33853

File: src/libraries/System.Diagnostics.Process/tests/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+[assembly: ActiveIssue("Disabled due to a runtime crash.", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
```
Comment (marek-safar) [inline src/libraries/System.Diagnostics.Process/tests/AssemblyInfo.cs:7]: This is serious and needs tracking issue

Comment (MaximLipnin) [reply src/libraries/System.Diagnostics.Process/tests/AssemblyInfo.cs:7]: Yeah, I didn't filed an issue in advanced in order to check that this disabling helps here.

**Review by akoeplinger — APPROVED** [review]
Great job!

File: src/libraries/System.Runtime.Serialization.Xml/tests/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+[assembly: ActiveIssue("Disabled due to multiple crashes.", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
```
Comment (akoeplinger) [inline src/libraries/System.Runtime.Serialization.Xml/tests/AssemblyInfo.cs:7]: @MaximLipnin missed this one during the review, please open an issue for it and replace the message with the link.

Comment (MaximLipnin) [reply src/libraries/System.Runtime.Serialization.Xml/tests/AssemblyInfo.cs:7]: oh, sorry, I missed it too. I'll update it with an actual issue.

Comment (MaximLipnin) [reply src/libraries/System.Runtime.Serialization.Xml/tests/AssemblyInfo.cs:7]: `System.Runtime.Serialization.Xml.Tests` has been enabled in https://github.com/dotnet/runtime/pull/34816. No issue is required.

**Pull request merged**